### DISCUSSION TITLE
Is There Somewhere I Can Fetch Initial App Data to Use Globally in an ISR App?

### DISCUSSION THREAD

**Discussion by darrylmorley** [vercel/next.js#25701] [Help]
Currently my app is making too many API calls (rate limited api) whilst using getStaticProps & getStaticPaths, to reduce these I would like to pull some initial data at the start of the build time and add that to react context.

It doesn't appear that this should be done in _app.js, where should I pull the initial data?

**Comment by lxsmnsyc** [comment]
If `_app` has a `getInitialData`, all of your pages will opt-out of Static Page Optimization.

`_document` won't allow you to get initial data as `_document` is rendered after the app has been rendered.

If you're looking for fetching the initial data during build time, have you looked into `getStaticProps`?

**Reply by darrylmorley** [reply]
Hi @LXSMNSYC Thanks for the response. I am using getStaticProps, getStaticPaths in my pages. I'm looking for somewhere to pull initial app data that can be shared across the whole app at the beginning of the build.

**Reply by lxsmnsyc** [reply]
There's no solution without trading off the SSO. You can, however, perform contextual data-fetching (e.g. SWR + Context) but it isn't SSR-friendly to do so.

**Reply by darrylmorley** [reply]
Yes, I thought that may be the case 😱. Although, I was hoping that there may be a workaround.

**Reply by arcataroger** [reply]
It's not the "Next" way of doing things, but I think you can use a script to download the JSON into your repo and import it as a static file into your pages.

Alternatively, you can cache your API endpoint using something like a Cloudflare Worker: https://developers.cloudflare.com/workers/examples/cache-using-fetch

**Reply by darrylmorley** [reply]
@arcataroger Thanks for the tip, I'll take a look at Cloudflare Workers.
